grakn-engine/src/main/java/ai/grakn/engine/GraknEngineServer.java

```diff
@@ -160,18 +158,20 @@ private void loadAndUnlock(Lock lock) {
 
     /**
      * Check in with the properties file to decide which type of task manager should be started
-     * @param inMemoryQueue
-     * @param redisCountStorage
+     * and return the TaskManager
+     * @param inMemoryQueue         True if running in memory
      * @param jedisPool
      */
     private TaskManager startTaskManager(
             boolean inMemoryQueue,
-            RedisCountStorage redisCountStorage,
-            Pool<Jedis> jedisPool, LockProvider lockProvider) {
+            Pool<Jedis> jedisPool,
+            LockProvider lockProvider) {
         TaskManager taskManager;
         if (!inMemoryQueue) {
             taskManager = new RedisTaskManager(engineId, prop, jedisPool, factory, lockProvider, metricRegistry);
         } else  {
+            // Redis storage for counts, in the RedisTaskManager it's created in consumers
+            RedisCountStorage redisCountStorage = RedisCountStorage.create(jedisPool);
```

Comment:
Good question! 
I preferred moving it to the consumers in this case. It seemed a bit like breaking the responsibilities of a generic task manager to pass it there. Instead it's created inside the consumers. It doesn't have any particular performance hit because the heavy object, the jedisPool, is passed as an argument and RedisCountStorage is just a class with convenience methods for the counts